main.tf

```
@@ -20,6 +20,9 @@ module "terraform_state_bucket" {
   logging_bucket = local.logging_bucket
 
   use_account_alias_prefix = false
+  bucket_key_enabled       = true
```

You defined the variable, but then you hard-code the value. I believe this should be
```suggestion
  bucket_key_enabled       = var.bucket_key_enabled
```

---

Oh jeez, good catch!